simphony/cuds/material.py

```diff
@@ -0,0 +1,31 @@
+import uuid
+
+from simphony.core.data_container import DataContainer
+
+
+class Material(object):
+    """A class that represents a material.
+
+    Class describes a material and its data.
+
+    Attributes
+    ----------
+    description : str
+        textual description of the material
+    uid : uuid.UUID
+        the id of the material. If None, then it is generated
+    data : DataContainer
+        data associated with this material
+
+    """
+    def __init__(self, uid=None, data=None, description=""):
```

Comment:
@nathanfranklin the unique id notion needs to be general to the project. You can surely add a separate notion of name with all the required machinery (query_by_name, ensure name is unique, etc.) but it should not bypass the uid notion. If I am not wrong (and I may be ;-)), this is used in the serialization process and the link with other objects, no?